src/v2/recommender.py

```diff
@@ -82,10 +90,19 @@ def _call(payload: InsightsRequest):
             )
             return json_response
 
-    def _get_insights_response(self, normalized_packages: NormalizedPackages):
+    def _get_insights_response(self, request: RecommenderRequest):
+        try:
+            insights_url = ECOSYSTEM_TO_INSIGHTS_URL[request.ecosystem]
+        except KeyError as e:
+            raise InsightsNotSupportedEcosystems from e
+
+        if not request.packages:
+            raise InsightsWithEmptyPackageException("Request package list is empty")
+        normalized_packages = NormalizedPackages(request.packages, request.ecosystem)
```

Comment:
This normalization was not applied for golang ecosystem. This check / handling is missing. Can you cross check the changes with golang ecosystem?

Reply:
normalization is needed only if it is going to be processed. for golang, the control won't reach here.